File: P383 Ransom Note/Solution.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        count = collections.Counter(magazine)
        for letter in ransomNote:
            count[letter] -= 1
            if count[letter] <0: return False
        return True

File: P383 Ransom Note/Solution.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int count[26];
        for (int i = 0; i < 26; i++)
            count[i] = 0;
        for (int i = 0; i < magazine.length(); i++) 
            count[magazine.at(i) - 97]++;
        for (int i = 0; i < ransomNote.length(); i++) 
            if (--count[ransomNote.at(i) - 97] < 0) 
                return false;
        return true;        
    }
};